File: src/main/java/com/horizen/transaction/Transaction.java
/*
package com.horizen.transaction;


import scorex.crypto.hash.Blake2b256;
import scorex.util.encode.Base16;

public abstract class Transaction implements scorex.core.transaction.Transaction
{

    @Override
    public String id() {
        return Base16.encode(Blake2b256.hash(messageToSign()));
    }

    @Override
    public abstract byte[] bytes();

    @Override
    public abstract byte[] messageToSign();

    @Override
    public abstract TransactionSerializer serializer();

    public abstract byte transactionTypeId();
}
*/

File: sdk/src/main/scala/io/horizen/transaction/Transaction.scala
package io.horizen.transaction

import io.horizen.transaction.exception.TransactionSemanticValidityException

abstract class Transaction
  extends sparkz.core.transaction.Transaction
{
  def transactionTypeId: Byte
  def version: Byte

  @throws[TransactionSemanticValidityException]
  def semanticValidity(): Unit

  def size(): Long
}
